### ISSUE TITLE
Document router, dispatcher, routing

### ISSUE THREAD

**Issue by matks** [PrestaShop/docs#994]

**Comment by kpodemski** [comment]
I think it's done here:
https://devdocs.prestashop.com/1.7/development/components/router-dispatcher/#router-and-dispatcher

right @matks ?

**Comment by matks** [comment]
So
- router ✅ 
- dispatcher ✅

**Comment by matks** [comment]
Routing: wdyt ? is it clear, for new PrestaShop users, how to create a new FO route ? a new BO legacy route ? a new BO Symfony route ?

Should we also document URL rewriting (how to use it and how it works internally) ?

**Comment by kpodemski** [comment]
@matks 

I don't think so, many developers don't know about `hookModuleRoutes` and that they can use it to create custom routing, for the Symfony routes we have some information already:
https://devdocs.prestashop.com/1.7/modules/concepts/controllers/admin-controllers/route-generation/#generate-a-symfony-route-from-a-module

> Should we also document URL rewriting (how to use it and how it works internally) ?

Hmmm, maybe for the legacy part? We have some information already https://devdocs.prestashop.com/1.7/modules/concepts/controllers/front-controllers/#front-controllers but no precise information on how PrestaShop can determine whether we want to display CMS, Module, or Product for example.
